### PR TITLE
Deprecate support for Ruby versions earlier than v2.0

### PR DESCRIPTION
**Pull request by floehopper**
This was based on #553.

This is long overdue. [Extended maintenance of Ruby v1.9.3 ended on 23 Feb 2015][1]. Actually dropping support will allow significant simplification of the code.

The immediate motivation is to come up with better parameter matching for keyword arguments and we've run into parsing issues when trying to do this in conjunction with continuing to support Ruby v1.9.

See #325.

[1]: https://www.ruby-lang.org/en/news/2014/01/10/ruby-1-9-3-will-end-on-2015/

### CODE REVIEW

**Comment by floehopper** [comment]
@wasabigeek While I was expanding the commit note, I noticed that the deprecation warning wasn't showing up in the CI build for Ruby v1.9. This seems to be because `lib/mocha.rb` is not automatically required when the `mocha` gem is included in a `Gemfile`. Moving the deprecation warning into `lib/mocha/ruby_version.rb` seems to fix the problem and I can now see the warning in [the build output](https://app.circleci.com/pipelines/github/freerange/mocha/165/workflows/05b908a1-d096-4dc0-a403-558e18ec0a3d/jobs/2322/parallel-runs/0/steps/0-108). I suspect that means that the previous deprecation warning may not have actually worked correctly, although perhaps something else has changed in the meantime. Anyway, I'm pretty tired, so I'd appreciate another pair of eyes on this before I merge it!

**Review by wasabigeek — APPROVED** [review]
This works for me, though I couldn't figure out how to install Ruby 1.9 and had to reverse this conditional instead 😄.

If I understand correctly though, this comes up more as a side effect of the chain of `require`s kicked off by `require 'mocha/mini_test'` (or similar) i.e. if somebody somehow manages to initiate a code path that doesn't `require 'mocha/ruby_version'`, they won't see the warning. I guess this probably won't be an issue given that `mocha/api` is the common entry-point and it will trigger the warning 👍

**Comment by floehopper** [comment]
@wasabigeek Thanks for looking at this - that's really helpful. I think I might try moving the deprecation warning into `mocha/api`. That makes more sense to me now.

**Comment by floehopper** [comment]
In the end I've decided to leave the deprecation warning in 'mocha/ruby_version', but to require that file from 'mocha/test_unit', 'mocha/minitest' & 'mocha/api' to cater for Test::Unit, Minitest & RSpec integration respectively.

**Pull request merged**
